lib/improver/tests/utilities/test_TemporalInterpolation.py

```diff
@@ -0,0 +1,234 @@
+        with self.assertRaisesRegex(ValueError, msg):
+            TemporalInterpolation(times=self.times).construct_time_list(
+                self.time_0, self.time_1)
+
+
+class Test_process(IrisTest):
+
+    """Test interpolation of cubes to intermediate times using the plugin."""
+
+    def setUp(self):
+        """Set up the test inputs."""
+        self.time_0 = datetime.datetime(2017, 11, 1, 3)
+        self.time_extra = datetime.datetime(2017, 11, 1, 6)
+        self.time_1 = datetime.datetime(2017, 11, 1, 9)
+        self.npoints = 10
+        data_time_0 = np.ones((self.npoints, self.npoints))
+        data_time_1 = np.ones((self.npoints, self.npoints)) * 7
+
+        cube_template = Cube(data_time_0, 'air_temperature', 'K')
+        cube_template.add_dim_coord(
+            DimCoord(np.linspace(-45.0, 45.0, self.npoints),
+                     'latitude', units='degrees'), 0)
+        cube_template.add_dim_coord(
+            DimCoord(np.linspace(120, 180, self.npoints),
+                     'longitude', units='degrees'), 1)
+        time_origin = "seconds since 1970-01-01 00:00:00"
+        calendar = "gregorian"
+        tunit = Unit(time_origin, calendar)
+
+        self.cube_time_0 = cube_template
+        self.cube_time_1 = cube_template.copy(data=data_time_1)
+        self.cube_time_0.add_aux_coord(DimCoord(self.time_0.timestamp(),
+                                                "time", units=tunit))
+        self.cube_time_1.add_aux_coord(DimCoord(self.time_1.timestamp(),
+                                                "time", units=tunit))
+
+    def test_valid_single_interpolation(self):
+        """Test interpolating to the mid point of the time range. Expect the
+        data to be half way between, and the time coordinate should be at
+        06Z November 11th 2017."""
+
+        expected_data = np.ones((self.npoints, self.npoints)) * 4
+        expected_time = (self.time_0 + timedelta(hours=3)).timestamp()
+        result, = TemporalInterpolation(interval_in_minutes=180).process(
+            self.cube_time_0, self.cube_time_1)
+
+        self.assertArrayEqual(expected_data, result.data)
+        self.assertEqual(result.coord('time').points, expected_time)
+
+    def test_valid_multiple_interpolations(self):
+        """Test interpolating to every hour between the two input cubes.
+        Check the data increments as expected and the time coordinates are also
+        set correctly.
+
+        NB Interpolation in iris is prone to float precision errors of order
+        10E-6, hence the need to use assert_almost_equal below."""
+
+        result = TemporalInterpolation(interval_in_minutes=60).process(
+            self.cube_time_0, self.cube_time_1)
+        for i, cube in enumerate(result):
+            expected_data = np.ones((self.npoints, self.npoints)) * i + 2
+            expected_time = (self.time_0 + timedelta(hours=(i+1))).timestamp()
+
+            np.testing.assert_almost_equal(expected_data, cube.data)
```

Comment:
Why are you using `np.testing.assert_almost_equal` rather than the normal `self.assertArrayAlmostEqual`?

Reply:
I tend to use the numpy ones as you have more options, but the normal ones work fine here. Changed.

Reply:
The documentation is here: https://docs.python.org/3.5/library/unittest.html#assert-methods
You can put optional values into the ArrayAlmostEqual test like you had in the numpy version